src/ui/groups_widgets.cc

```diff
@@ -772,6 +773,57 @@ void DictGroupsWidget::addAutoGroups()
   }
 }
 
+
+void DictGroupsWidget::addAutoGroupsByFolders()
+{
+  if ( activeDicts->empty() ) {
+    return;
+  }
+
+  if ( QMessageBox::information( this,
+                                 tr( "Confirmation" ),
+                                 tr( "Are you sure you want to generate a set of groups "
+                                     "based on containing folders?" ),
+                                 QMessageBox::Yes,
+                                 QMessageBox::Cancel )
+       != QMessageBox::Yes ) {
+    return;
+  }
+
+  QMultiMap< QString, sptr< Dictionary::Class > > dictMap;
+
+  for ( const auto & dict : *activeDicts ) {
+    QString dictFolder = dict->getContainingFolder();
+
+    if ( dictFolder.isEmpty() ) {
+      continue;
+    }
+
+    QDir dir = dictFolder;
+    if ( dir.cdUp() ) {
+      dictMap.insert( dir.dirName(), dict );
+    }
+    else {
+      qWarning() << "Cannot auto group because parent folder cannot be reached: " << dir;
+      continue;
+    }
```

Comment:
on this condition ,  what about    `dictMap.insert(dictFolder,dict)`

Reply:
This is a philosophy problem.

Simplicity vs Pragmatic.

I think without knowing what's the actual algorithms behind "Auto Group by Folder", a user will try to understand the functionality from the result (since nobody read doc).

If we add the current container folder for single files, then the deciphering of the actual algorithm used will be much harder, because they have to deduce two rules instead of one.

So I don't want to add that :sweat_smile:.

Reply:
Should we add a document about this folder group alogrithm in the `howto`?

Reply:
I will write some new content on the website in another PR explaining this.

Reply:
Or we make a new build after this, and I write a post and see how they will question this functionality.

Reply:
another PR is ok